### DISCUSSION TITLE
Running integration tests locally fails

### DISCUSSION THREAD

**Discussion by Blargian** [ClickHouse/ClickHouse#58882] [Q&A]
I'm new to contributing to ClickHouse and I'm trying to run some integration tests which are now failing after a change in the codebase but ran into some trouble getting integration tests to run. 

From `tests/integration`, after following `README.md` to install the dependencies I run `pytest -k test_block_based_formats_1` to run the test which CI reports is failing, and I get this error:

```
ERROR test_storage_kafka/test.py::test_block_based_formats_1 - FileNotFoundError: [Errno 2] No such file or directory: '/etc/clickhouse-server/config.xml'
ERROR test_storage_rabbitmq/test.py::test_block_based_formats_1 - FileNotFoundError: [Errno 2] No such file or directory: '/etc/clickhouse-server/config.xml'
```
I checked for `etc/clickhouse-server/config.xml` however I don't find it anywhere in the repository. I have however when running ClickHouse locally been using the `config.xml` file located at `programs/server/`. 

So my question is:

- where can I find `etc/clickhouse-server/config.xml` and more generally what is the difference between this config file and the one located at `programs/server`?
- If this file does not exist for me locally, how can I instruct pytest to use the config file at `programs/server`?  (or another config file, if this is not the correct one to use)

**Comment by Algunenano** [comment]
Here is an example of how I'd run it locally with my paths (note that this a single bash line)
```sh
CLICKHOUSE_TESTS_CLIENT_BIN_PATH=/mnt/ch/ClickHouse/build/programs/clickhouse \
CLICKHOUSE_TESTS_SERVER_BIN_PATH=/mnt/ch/ClickHouse/build/programs/clickhouse \ 
CLICKHOUSE_TESTS_BASE_CONFIG_DIR=/mnt/ch/ClickHouse/programs/server/ \
./runner \
--binary /mnt/ch/ClickHouse/build/programs/clickhouse \
--odbc-bridge-binary /mnt/ch/ClickHouse/build/programs/clickhouse-odbc-bridge \
--base-configs-dir /mnt/ch/ClickHouse/programs/server/ \
'test_storage_kafka/test.py::test_block_based_formats_1'
```

You'll need docker running. There might be simpler ways, but that's how I usually launch them. Hope that helps

**Comment by Blargian** [comment]
@Algunenano thanks, that helps! Managed to get it working with slight modification to your command by changing the path for `--odbc-bridge-binary` and adding an argument for `--library-bridge-binary`:

```
CLICKHOUSE_TESTS_CLIENT_BIN_PATH=/home/shaun/Desktop/ClickHouse/build/programs/clickhouse \
CLICKHOUSE_TESTS_SERVER_BIN_PATH=/home/shaun/Desktop/ClickHouse/build/programs/clickhouse \ 
CLICKHOUSE_TESTS_BASE_CONFIG_DIR=/home/shaun/Desktop/ClickHouse/programs/server/ \
./runner \
--binary /home/shaun/Desktop/ClickHouse/build/programs/clickhouse \
--odbc-bridge-binary /home/shaun/Desktop/ClickHouse/build/programs/odbc-bridge \
--library-bridge-binary /home/shaun/Desktop/ClickHouse/build/programs/library-bridge \
--base-configs-dir /home/shaun/Desktop/ClickHouse/programs/server/ \
'test_storage_kafka/test.py::test_block_based_formats_1'
```
